src/Components/App/App.js

```diff
@@ -27,130 +29,181 @@ class App extends Component {
       music: [],
       podcasts: [],
       boardGames: [],
-      cardGames: []
-    }
+      cardGames: [],
+    };
+  this.baseState = this.state
   }
 
   getActivityData = async (name) => {
     if (name.target) {
-      name = name.target.id
+      name = name.target.id;
     }
-    if (name.includes('games')) {
-      name = name.replace(/ games/gi, 'Games')
+    if (name.includes("games")) {
+      name = name.replace(/ games/gi, "Games");
     }
     let promise;
     try {
-      if (name === 'movies') {
+      if (name === "movies") {
         promise = await getAllMovies();
       }
-      if (name === 'boardGames') {
+      if (name === "boardGames") {
         promise = await getAllBoardGames();
       }
-      if (name === 'cardGames') {
+      if (name === "cardGames") {
         promise = await getAllCardGames();
       }
-      if (name === 'music') {
+      if (name === "music") {
         promise = await getAllMusic();
       }
-      if (name === 'podcasts') {
+      if (name === "podcasts") {
         promise = await getAllPodcasts();
       }
-      this.setState({[name]: promise})
+      this.setState({ [name]: promise });
     } catch (error) {
-      console.log(error)
+      console.log(error);
     }
+  };
+
+  resetState = ()=>{
+    this.setState(this.baseState)
+  }
+  setHistory = () =>{
+    this.setState({
+      history:[...this.state.history, this.state],
+    })
+  }
+  goBack = ()=>{
+    this.setState(this.state.history[this.state.history.length - 1])
   }
```

Comment:
These are all new functions to help keep state consistent across varying user stories

`resetState` is used when a user selects the home button to get a clean slate
`setHistory` is used to keep note of previous states (updated on next button)
`goBack` is utilized by the back buttons to go to the most previous state

Reply:
I love how organized this is! Thank you for putting so much thought into the state here in App.